server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -310,17 +312,25 @@ private void searchLeaf(LeafReaderContext ctx, Weight weight, Collector collecto
             return;
         }
 
-        cancellable.checkCancelled();
-        weight = wrapWeight(weight);
-        // See please https://github.com/apache/lucene/pull/964
-        collector.setWeight(weight);
         final LeafCollector leafCollector;
         try {
+            cancellable.checkCancelled();
+            weight = wrapWeight(weight);
+            // See please https://github.com/apache/lucene/pull/964
+            collector.setWeight(weight);
             leafCollector = collector.getLeafCollector(ctx);
         } catch (CollectionTerminatedException e) {
             // there is no doc of interest in this reader context
             // continue with the following leaf
             return;
+        } catch (QueryPhase.TimeExceededException e) {
+            // Swallow timeout exception and return
+            if (searchContext.request().allowPartialSearchResults() == false) {
```

Comment:
As of today, `searchContext` could be `null`, so this won't work in all cases.

Reply:
I believe this is the same `searchContext` that is already being used in the query phase here: https://github.com/opensearch-project/OpenSearch/blob/068404ed1cef5e1d023816d57054aef98a7aa343/server/src/main/java/org/opensearch/search/query/ConcurrentQueryPhaseSearcher.java#L85-L92 So it seems to me that we are already expecting that it is not null by this point.

I supposed we could add a null check and then rethrow the exception if `searchContext` is `null`, but could you help me understand how `searchContext` could end up being null at this point?

Reply:
Check please the `ContextIndexSearcher` constructors, we should probably remove the one which does not accepts `searchContext`

Reply:
https://github.com/opensearch-project/OpenSearch/blob/0173f13b50067ba3c85a0ab229a18ab3ac582302/server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java#L108-L127

This is the only place that I could find with `searchContext` being passed in as `null`, and this constructor is only used in tests.
I'm going to add a null check to be safe and leave a comment about how that constructor is only used for testing.

Reply:
I do like the idea of removing the constructor completely so we can be confident there's no null `searchContext`, but I think that can be it's own separate issue since that's going to span a lot of other classes.

Reply:
> I do like the idea of removing the constructor completely so we can be confident there's no null searchContext, but I think that can be it's own separate issue since that's going to span a lot of other classes.

Please remove it, I kept it for bwc at the time, the changes would be only in tests (afaik)